ci: set PR base for codeql workflow

tklauser opened the pull request:
This fixes the following error on scheduled runs:

    This action requires 'base' input to be configured or 'repository.default_branch' to be set in the event payload"

This makes sure only code changed by the CL is flagged by CodeQL,
but not existing issues in the repo.

Moreover, run the paths-filter action only on pull_request events. This
should fix scheduled run of the workflow.

Noticed while investigating #18282.

pchaigno commented on .github/workflows/lint-codeql.yaml:
```diff
@@ -23,10 +23,19 @@ jobs:
         id: go-changes
         with:
+          base: ${{ steps.pr.outputs.base }}
+          ref: ${{ steps.pr.outputs.head }}
```
For a workflow run triggered by a pull request, I think `github.event.pull_request.{head,base}.sha` are enough here (`github.event.issue` is probably not available on that trigger anyway). But this workflow can also be triggered on schedule on by push to the master branch; shouldn't we account for that as well in these values?

tklauser replied:
Good point. I guess we only want to compare against the base for `github.event_name == pull_request` (i.e. to only check code changes), but run against the full code base on scheduled runs. The dorny/paths-filter action seems to require `base` or `repository.default_branch` being set for scheduled runs as well, thus the error. So I guess we just make that action conditional on `github.event_name == pull_request`.

pchaigno replied:
> So I guess we just make that action conditional on `github.event_name == pull_request`.

Yep, I think that would be fine.

tklauser committed:
ci: set PR base and ref for code codeql workflow

This fixes the following error on scheduled runs:

    This action requires 'base' input to be configured or 'repository.default_branch' to be set in the event payload"

This also makes sure only code changed by the CL is flagged by CodeQL,
but not existing issues in the repo.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
ci: run CodeQL action only on PRs against master

This fixes the following reported issue [1]:

    1 issue was detected with this workflow: Please make sure that every
    branch in on.pull_request is also in on.push so that Code Scanning
    can compare pull requests against the state of the base branch.

[1] https://github.com/cilium/cilium/actions/runs/1591471691

Note that the release branches have their own version of this workflow
and on.{pull_request,push}.branches need to be changed accordingly.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

pchaigno commented on .github/workflows/lint-codeql.yaml:
```diff
@@ -1,7 +1,9 @@
-  pull_request: {}
+  pull_request:
+    branches:
+      - master
```
Don't we also want it to run for backport PRs?

tklauser replied:
It looks like the release branches have their own version of this workflow with `on.push.branches` set to the respective branch. Since this is triggered `on.pull_request` (not `on.pull_request_event`), I think the workflow from the corresponding PR branch is run. Or did I mix something up there?

pchaigno replied:
> Since this is triggered on.pull_request (not on.pull_request_event), I think the workflow from the corresponding PR branch is run.

Forgot about that :facepalm:
